apps/examples/wifi_manager_sample/wm_test_interop.c

```diff
@@ -0,0 +1,679 @@
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <semaphore.h>
+#include <errno.h>
+#include <wifi_manager/wifi_manager.h>
+#include "wm_test.h"
+#include "wm_test_log.h"
+
+static sem_t g_wio_sem;
+
+#define TAG "[WTIO]"
+
+#define DELIMITER ","
+#define DELI_LEN 1
+#define WIFI_INTEROP_BUFSIZE 128
+#define WIFI_INTEROP_PATH "/mnt/"
+#define WIFI_INTEROP_FILENAME "wm_aplist.conf"
+#define WIFI_INTEROP_RESULT "wm_result.conf"
+
+#define WIO_TEST_SIGNAL								\
+	do {											\
+		sem_post(&g_wio_sem);			\
+		WT_LOG(TAG, "send signal");	\
+	} while (0)
+
+#define WIO_TEST_WAIT								\
+	do {											\
+		WT_LOG(TAG, "wait signal");	\
+		sem_wait(&g_wio_sem);				\
+	} while (0)
+
+
+
+typedef enum {
+	SSID,
+	SECURITY,
+	PASSPHRASE
+} wio_config_e;
+
+int g_ap_found_status;
+int g_current_ap_idx;
+int g_connect_status;
+int g_wm_init_flag;
+
+#define MAX_INTEROP_AP_CONFIG_COUNT 10
+#define MAX_INTEROP_SECURITY_LEN 10
+
+typedef struct {
+	wifi_manager_ap_config_s ap_config;
+	char security[MAX_INTEROP_SECURITY_LEN];
+	int scan_result;
+	int connect_result;
+} interop_ap_info_s;
+
+typedef struct {
+	int ap_count;
+	interop_ap_info_s ap_info[MAX_INTEROP_AP_CONFIG_COUNT];
+} interop_ap_config_list_s;
+
+static interop_ap_config_list_s g_apconfig_list;
+
+
+static void print_wifi_ap_profile(wifi_manager_ap_config_s *config)
+{
+	if(config == NULL)
+		return;
+
+	WT_LOG(TAG, "========================================================");
+	WT_LOG(TAG, "SSID[%s] passphrase[%s] auth type:[%d] crypto type: [%d]",
+		config->ssid, config->passphrase, config->ap_auth_type, config->ap_crypto_type);
+	WT_LOG(TAG, "========================================================");
+}
+
+static void wm_sta_connected(wifi_manager_result_e res)
+{
+	WT_LOG(TAG, "res(%d)", res);
+
+	if(res == WIFI_MANAGER_SUCCESS)
+		g_connect_status = 1;
+	else
+		g_connect_status = 0;
+
+	WIO_TEST_SIGNAL;
+}
+
+static void wm_sta_disconnected(wifi_manager_disconnect_e disconn)
+{
+	WT_LOG(TAG, "res(%d)", disconn);
+	g_connect_status = 0;
+	WIO_TEST_SIGNAL;
+}
+
+static void wm_scan_done(wifi_manager_scan_info_s **scan_result, wifi_manager_scan_result_e res)
+{
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted soon eventually as you exit this function.
+	 */
+	if (scan_result == NULL) {
+		WIO_TEST_SIGNAL;
+		return;
+	}
+	wifi_manager_scan_info_s *wifi_scan_iter = *scan_result;
+	while (wifi_scan_iter != NULL) {
+		WT_LOG(TAG, "WiFi AP SSID: %-25s, BSSID: %-20s, Rssi: %d, Auth: %d, Crypto: %d",
+				wifi_scan_iter->ssid, wifi_scan_iter->bssid, wifi_scan_iter->rssi,
+				wifi_scan_iter->ap_auth_type, wifi_scan_iter->ap_crypto_type);
+
+		interop_ap_info_s *ap_info = (g_apconfig_list.ap_info + g_current_ap_idx);
+
+		//check AP
+		if( strcmp(wifi_scan_iter->ssid, ap_info->ap_config.ssid) == 0 ) {
```

Comment:
Please insert a space after if and before (.

Reply:
It will better to use strncmp instead of strcmp ?